phc/easy/query/__init__.py

```diff
@@ -216,13 +220,44 @@ def execute_paging_api(
                 }
             )
         """
-        return with_progress(
+
+        auth = Auth(auth_args)
+
+        params = clean_params(params)
+        path = path.replace(":project_id", auth.project_id)
+        query = {"path": path, "method": http_verb, "params": params}
+
+        if all_results and page_size is None:
+            # Default to 100 if not provided but getting all results
+            page_size = 100
+
+        if log:
+            print(json.dumps(query, indent=4))
+
+        use_cache = (
+            (not ignore_cache)
+            and (not raw)
+            and all_results
+            and (max_pages is None)
+        )
+
+        if use_cache and APICache.does_cache_for_query_exist(query):
+            return APICache.load_cache_for_fhir_dsl(query)
+
+        callback = (
+            APICache.build_cache_callback(query, transform, nested_key=None)
+            if use_cache
+            else None
+        )
```

Comment:
This caching code closely parallels the `Query.execute_fhir_dsl_with_options` for fetching FSS entities.